From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: 404Setup <153366651+404Setup@users.noreply.github.com>
Date: Wed, 12 Feb 2025 22:53:52 +0800
Subject: [PATCH] Divine: Fix MC-172801

Original code by NONPLAYT
You can find the original code on https://github.com/DivineMC/DivineMC

Original post on Mojira: https://bugs.mojang.com/browse/MC-172801

diff --git a/net/minecraft/world/entity/LivingEntity.java b/net/minecraft/world/entity/LivingEntity.java
index 0a7c630e1c73d97bc8f7f444a3a4878b90af0a83..4eb5218b2257e7c35d421b5bcc61287ec031bf66 100644
--- a/net/minecraft/world/entity/LivingEntity.java
+++ b/net/minecraft/world/entity/LivingEntity.java
@@ -3131,7 +3131,13 @@ public abstract class LivingEntity extends Entity implements Attackable {
     }
 
     protected float getFlyingSpeed() {
-        return this.getControllingPassenger() instanceof net.minecraft.world.entity.player.Player ? this.getSpeed() * 0.1F : 0.02F;
+        // DivineMC start - Fix MC-172801
+        float flyingSpeed = 0.02F;
+        if (this.getAttributes().hasAttribute(Attributes.FLYING_SPEED)) {
+            flyingSpeed = (float) (this.getAttribute(Attributes.FLYING_SPEED).getValue() * 0.049999999254942D);
+        }
+        return this.getControllingPassenger() instanceof net.minecraft.world.entity.player.Player ? this.getSpeed() * 0.1F : flyingSpeed;
+        // DivineMC end
     }
 
     public float getSpeed() {
